url74c52881d6_AndrewBourgeois_JGWCAConstants_tgz-pJ8-be_demmel_fun_jgwcaconstants_Tutorial4J8/src/be/demmel/fun/jgwcaconstants/Version.java

```diff
@@ -0,0 +1,67 @@
+package be.demmel.fun.jgwcaconstants;
+
+import java.io.InputStream;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Version {
+
+    private static final Logger logger = LoggerFactory.getLogger(Version.class);
+    // Build the static information when the class is loaded
+    private static final String FULL_VERSION, VERSION;
+    // The path inside the library where the properties file generated by Maven is located
+    //TODO: find a way to fix the location of the effective POM and .properties so that this Version class can be reused in every project
+    private static final String POM_PATH = "/META-INF/maven/be.demmel.fun/jgwcaconstants/pom.properties";
+    // The message that will be printed when any kind of error occurs during the retrieval of the information
+    private static final String ERROR_MESSAGE = "[!ERROR!]";
+    
+    static {
+        String fullVersion = null;
+        
+        StringBuilder sb = new StringBuilder();
+        try {
+            Properties props = loadProperties();
+            sb.append(props.get("artifactId")).append(" ");
+            sb.append(props.get("version"));
+            fullVersion = sb.toString();
+        } catch (Exception ex) {
+            logger.error("An unexpected problem occured loading {}:", POM_PATH,ex);
+            fullVersion = ERROR_MESSAGE;
+        }
+        
+        FULL_VERSION = fullVersion;
+        
+        String version = null;
+        
+        try {
+            Properties props = loadProperties();
+            version = props.get("version").toString();
+        } catch (Exception ex) {
+            logger.error("An unexpected problem occured loading the Maven properties of the module: ", ex);
+            version = ERROR_MESSAGE;
+        }
+        
+        VERSION = version;
+    }
+
+    public static void main(String[] args) {
+        System.out.println(FULL_VERSION);
+    }
+
+    public static String getVersion() {
+        return VERSION;
+    }
+    
+    // Loads the properties generated by Maven in the build process
+    private static Properties loadProperties() throws Exception {
+        InputStream stream = Version.class.getResourceAsStream(POM_PATH);
+        Properties props = new Properties();
+        try {
+            props.load(stream);
+            return props;
+        } catch (Exception ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)